trillian/ctfe/handlers.go

```diff
@@ -231,8 +231,10 @@ type logInfo struct {
 	validationOpts CertValidationOpts
 	// rpcClient is the client used to communicate with the trillian backend
 	rpcClient trillian.TrillianLogClient
-	// signer signs objects
+	// signer signs objects (e.g. STHs, SCTs) for regular logs
```

Comment:
It would be nice if we could unify the signer + provider api somehow. Less to go wrong then. This might be hard?

Reply:
Yeah, I started doing a sort of `LatestSTHGetter` interface with a `GetSTH(ctx)` method that would have an implementation for both types of logs, but then I realized the quota bit around it, so it should actually be a `GetSTH(ctx, quotaUser)` which looked less nice. But after another though seems not too bad.

Reply:
Hm, could we bake quota user to the `Context`?

Reply:
It's sort of a fixed parameter related to processing the request. It could work.

Reply:
Btw, the signer is also used for SCTs, so this line would still be here.

Reply:
Introduced an `STHGetter` interface, PTAL.